### PR TITLE
Migrate to v4 version of aws provider

### PR DESCRIPTION
**Pull request by dolohow**
v4 deprecates policy argument in `aws_s3_bucket` resource[1], therefore users who would like to have this functionality should create `aws_s3_bucket_policy`.

[1] https://registry.terraform.io/providers/hashicorp/aws/latest/docs/guides/version-4-upgrade#policy-argument

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -63,7 +63,6 @@ resource "aws_s3_bucket" "default" {
   provider      = aws.virginia
   bucket        = var.domain_name
   force_destroy = var.force_destroy
-  policy        = var.policy
```
Comment (nagelflorian) [inline main.tf:66]: Have you considered adding a `aws_s3_bucket_policy` resource to avoid dropping the bucket policy altogether?

**Review by nagelflorian — CHANGES_REQUESTED** [review]
Thanks for your contribution.

Can we perhaps add a `aws_s3_bucket_policy` resource to keep the bucket policy functionality like [shown in the upgrade guide](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/guides/version-4-upgrade#migrating-to-aws_s3_bucket_policy).

File: main.tf (hunk 1 above)
Comment (dolohow) [reply main.tf:66]: We are not dropping it, you can easily enable it by creating `aws_s3_bucket_policy` outside this module and using `bucket_id` to link policy with it.  There is no point IMHO to pass JSON as a variable just to create this.  Also, someone might want to create two policies or something.  It is a breaking change, though, but not a big one, and everyone who would like to consider switching to v4 aws provider will need to put work anyway.

Does it make sense?

**Review by nagelflorian — APPROVED** [review]

**Comment by nagelflorian** [comment]
@dolohow thanks for your contribution and your patience. Sorry, that this one took a lot longer to get back to.

**Comment by nagelflorian** [comment]
Released in [v1.6.0](https://github.com/nagelflorian/terraform-aws-s3-review-apps/tree/v1.6.0)

**Pull request merged**